rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -604,5 +572,161 @@
+            </perform>
+        </rule>
+        <rule id="hibernate4-00035">
+            <when>
+                <javaclass references="org.hibernate.id.Configurable">
+                    <location>IMPLEMENTS_TYPE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Interface org.hibernate.id.Configurable changed method parameter in configure method in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Hibernate 5 changed the signature of `org.hibernate.id.Configurable#configure` method to accept `ServiceRegistry` rather than just `Dialect`</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#changedmoved-contracts"
+                        title="Hibernate 5 Changed or Moved Contracts" />
+                     <link
+                        href="https://docs.jboss.org/hibernate/orm/5.0/javadocs/org/hibernate/id/Configurable.html#configure-org.hibernate.type.Type-java.util.Properties-org.hibernate.service.ServiceRegistry-"
+                        title="Hibernate 5 Configurable API" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate4-00036">
+            <when>
+                <javaclass references="org.hibernate.id.PersistentIdentifierGenerator">
+                    <location>IMPLEMENTS_TYPE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="PersistentIdentifierGenerator implementations need to implement ExportableProducer#registerExportables" effort="1" severity="mandatory">
+                    <message>Since Hibernate 5, The `org.hibernate.id.PersistentIdentifierGenerator` extends new `org.hibernate.boot.model.relational.ExportableProducer` interface which will affect all `org.hibernate.id.PersistentIdentifierGenerator` implementations.
+                    Therefore you need to implement additionally `ExportableProducer#registerExportables(org.hibernate.boot.model.relational.Database database)` method.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#changedmoved-contracts"
+                        title="Hibernate 5 Changed or Moved Contracts" />
+                     <link
+                        href="https://docs.jboss.org/hibernate/orm/5.0/javadocs/org/hibernate/boot/model/relational/ExportableProducer.html#registerExportables-org.hibernate.boot.model.relational.Database-"
+                        title="Hibernate 5 ExportableProducer API" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hibernate4-00037">
+            <when>
+                <javaclass references="org.hibernate.envers.configuration.spi.AuditConfiguration" />
```

Comment:
Should be `org.hibernate.envers.configuration.AuditConfiguration` (and below too)